table/update_schema.go

```diff
@@ -0,0 +1,834 @@
+func (us *UpdateSchema) AllowIncompatibleChanges() *UpdateSchema {
+	us.allowIncompatibleChanges = true
+	return us
+}
+
+// SetCaseSensitive controls case sensitivity for field lookups
+func (us *UpdateSchema) SetCaseSensitive(caseSensitive bool) *UpdateSchema {
+	us.caseSensitive = caseSensitive
+	return us
+}
+
+// AddColumn queues an add column operation - validation deferred until Build is called
+func (us *UpdateSchema) AddColumn(path []string, required bool, dataType iceberg.Type, doc string, initialDefault any) *UpdateSchema {
+	op := &addColumnOp{
+		path:           path,
+		required:       required,
+		dataType:       dataType,
+		doc:            doc,
+		initialDefault: initialDefault,
+	}
+	us.operations = append(us.operations, op)
+	return us
+}
+
+// UpdateColumn queues an update column operation - validation deferred until Build is called
+func (us *UpdateSchema) UpdateColumn(path []string, updates ColumnUpdate) *UpdateSchema {
+	op := &updateColumnOp{
+		path:    path,
+		updates: updates,
+	}
+	us.operations = append(us.operations, op)
+	return us
+}
+
+// DeleteColumn queues a delete column operation - validation deferred until Build is called
+func (us *UpdateSchema) DeleteColumn(path []string) *UpdateSchema {
+	op := &deleteColumnOp{
+		path: path,
+	}
+	us.operations = append(us.operations, op)
+	return us
+}
+
+// Move queues a move column operation - validation deferred until Build is called
+func (us *UpdateSchema) Move(columnToMove, referenceColumn []string, op moveOp) *UpdateSchema {
+	moveOp := &moveColumnOp{
+		columnToMove:    columnToMove,
+		referenceColumn: referenceColumn,
+		op:              op,
+	}
+	us.operations = append(us.operations, moveOp)
+	return us
+}
+
+/// Operation Methods
+
+// Reset clears all queued operations and errors - validation deferred until Build is called
+func (us *UpdateSchema) Reset() *UpdateSchema {
+	us.operations = us.operations[:0]
+	us.errors = us.errors[:0]
+	return us
+}
+
+// RemoveLastOperation removes the most recently added operation
+func (us *UpdateSchema) RemoveLastOperation() *UpdateSchema {
+	if len(us.operations) > 0 {
+		us.operations = us.operations[:len(us.operations)-1]
+	}
+	return us
+}
+
+// GetQueuedOperations returns a copy of the queued operations
+func (us *UpdateSchema) GetQueuedOperations() []string {
+	result := make([]string, len(us.operations))
+	for i, op := range us.operations {
+		result[i] = op.String()
+	}
+	return result
+}
+
+// Validate runs all deferred validations for operations without building
+// basically just checks if the operations are valid
+func (us *UpdateSchema) Validate() error {
+	// Clear the previous errors
+	// because if validation is called multiple times, the errors will accumulate even after removing operations causing error
+	us.errors = us.errors[:0]
+
+	settings := &validationSettings{
+		allowIncompatibleChanges: us.allowIncompatibleChanges,
+		caseSensitive:            us.caseSensitive,
+	}
```

Comment:
why use a pointer for this? just take `validationSettings` instead of a `*validationSettings`